external-testing/README.md

```diff
@@ -0,0 +1,65 @@
+# External Testing
+
+Let's talk about testing process against external resources. What does that
+mean? Well, talking to GitHub API, COPR API, running `fedpkg`, cloning
+repositories...
+
+
+## Use cases
+
+* Interact with an external API (such as GitHub, COPR, Pagure, FAS, k8s)
+  * Authenticated vs. unauthenticated calls.
+* Clone a git repository.
+* Construct a git repository.
+* Run a command to perform an action.
+  * Changes happening only locally.
+  * RO access to a remote service.
+  * R/W access to a remote service.
+* Combination of multiple use cases into a single test case.
+
+
+## Solutions
+
+* flexmock
+
+* pytest fixtures
+
+* requre
+
+* VCRpy
+
+* Betamax
+
+
+## VCRpy vs. requre
+
+| A feature | VCRpy | requre |
+|-----------|-------|--------|
+| Record & replay for python requests | ✓ | ✓ |
+| Match on call stack | ✘ | ✓ |
+| Match on URL & request type | ✓ | ✘ |
+| Calling the same URL twice works as expected | ✓ | ✘ |
+| Record & replay for arbitrary functions | ✘ | ✓ |
+| Filtering sensitive data out of recorded responses | ✓ [1] | ✓ [2] |
+| Easy to use | ✓ [3] | ✘ [4] |
```

Comment:
depends :-) jup, vrcpy is easier for easier testcases, when you can decorate code directly for requrest module and match specific URL to record, but not in case when you would like to work transparently to python execution.

Reply:
I was able to start using vcrpy in minutes. With requre, it took me tenths of minutes and in the end I had to ask you for help. I'm sorry, no, requre is not easy to use.